application-ckeditor-plugins/src/main/resources/xwiki-macro/macroEditor.js

```diff
@@ -57,10 +58,28 @@ define('macroService', ['jquery'], function($) {
       });
     }
     return deferred.promise();
+  },
+
+  installMacro = function(extensionId, extensionVersion) {
+    var deferred = $.Deferred();
+
+    var url = new XWiki.Document('MacroService', 'CKEditor').getURL('get', $.param({
+      outputSyntax: 'plain',
+      language: $('html').attr('lang')
+    }));
+    $.get(url, {data: 'install', extensionId: extensionId, extensionVersion: extensionVersion, namespace: namespace})
+      .done(function(jobStatus) {
+        deferred.resolve(jobStatus);
+      }).fail(function(jobStatus) {
+        deferred.reject.apply(deferred, arguments);
```

Comment:
```suggestion
      }).fail(function() {
        deferred.reject.apply(deferred, arguments);
```

I assume there are additional arguments besides ``jobStatus`` that you want to pass to ``reject`` (otherwise you could have called it directly), so I'd rather not specify the parameters, otherwise it's misleading (took me some time to understand why you don't call reject ``directly``).